### PR TITLE
Fixed android warnings in _studio/mfx lib

### PR DESCRIPTION
**Pull request by Losev-Ivan**

### CODE REVIEW

File: _studio/mfx_lib/shared/src/libmfxsw_pak.cpp [hunk 1]
```diff
@@ -38,8 +38,10 @@
 template<>
 VideoPAK* _mfxSession::Create<VideoPAK>(mfxVideoParam& par)
 {
-    VideoPAK* pPAK = nullptr;
+#if defined(MFX_VA_LINUX) && defined(MFX_ENABLE_H264_VIDEO_ENCODE_HW) && defined(MFX_ENABLE_H264_VIDEO_FEI_PAK)
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/src/libmfxsw_pak.cpp]: I suggest to directly use m_pCore.get() below, i.e.:
`pPAK = (VideoPAK*) new VideoPAK_PAK(m_pCore.get(), &mfxRes);`
Less macros the better...

File: _studio/mfx_lib/shared/src/libmfxsw_enc.cpp [hunk 2]
```diff
@@ -46,8 +46,10 @@
 template<>
 VideoENC* _mfxSession::Create<VideoENC>(mfxVideoParam& par)
 {
-    VideoENC* pENC = nullptr;
+#if ( defined(MFX_ENABLE_LA_H264_VIDEO_HW) || defined(MFX_ENABLE_H264_VIDEO_FEI_PREENC) || defined(MFX_ENABLE_H264_VIDEO_FEI_ENC))
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/src/libmfxsw_enc.cpp]: I suggest to directly use m_pCore.get() below, i.e.:
pPAK = (VideoENC*) new VideoENC_ENC(m_pCore.get(), &mfxRes);
Less macros the better...

File: _studio/mfx_lib/shared/src/mfx_session.cpp [hunk 3]
```diff
@@ -408,7 +408,7 @@ CORE_FUNC_IMPL(IncreaseReference, (mfxHDL pthis, mfxFrameData *fd), (fd))
 CORE_FUNC_IMPL(DecreaseReference, (mfxHDL pthis, mfxFrameData *fd), (fd))
 CORE_FUNC_IMPL(CopyFrame, (mfxHDL pthis, mfxFrameSurface1 *dst, mfxFrameSurface1 *src), (dst, src))
 CORE_FUNC_IMPL(CopyBuffer, (mfxHDL pthis, mfxU8 *dst, mfxU32 dst_size, mfxFrameSurface1 *src), (dst, dst_size, src))
-CORE_FUNC_IMPL(CopyFrameEx, (mfxHDL pthis, mfxFrameSurface1 *dst, mfxU16  dstMemType, mfxFrameSurface1 *src, mfxU16  srcMemType), (dst, dstMemType, src, srcMemType))
+//CORE_FUNC_IMPL(CopyFrameEx, (mfxHDL pthis, mfxFrameSurface1 *dst, mfxU16  dstMemType, mfxFrameSurface1 *src, mfxU16  srcMemType), (dst, dstMemType, src, srcMemType))
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/src/mfx_session.cpp]: goes to separate change an PR! This needs to be discussed.

File: _studio/mfx_lib/cmrt_cross_platform/include/cmrt_cross_platform.h [hunk 4]
```diff
@@ -52,6 +52,9 @@
 #if defined(__clang__)
   #pragma clang diagnostic push
   #pragma clang diagnostic ignored "-Wnon-virtual-dtor"
+
```
Comment (dvrogozh) [inline _studio/mfx_lib/cmrt_cross_platform/include/cmrt_cross_platform.h]: Again, change in this file, in this file _alone_, should be placed in a separate commit!

File: _studio/shared/asc/include/asc_defs.h [hunk 5]
```diff
@@ -59,6 +59,10 @@
     #define ASC_PRINTF(...)     printf(__VA_ARGS__)
     #define ASC_FPRINTF(...)    fprintf(__VA_ARGS__)
     #define ASC_FFLUSH(x)       fflush(x)
+#elif defined(ANDROID)
```
Comment (dvrogozh) [inline _studio/shared/asc/include/asc_defs.h]: Again, this is separate bug-fix commit for Android.

File: _studio/mfx_lib/shared/src/mfx_session.cpp (hunk 3 above)
Comment (Losev-Ivan) [reply _studio/mfx_lib/shared/src/mfx_session.cpp]: https://github.com/Intel-Media-SDK/MediaSDK/pull/614

File: _studio/shared/asc/include/asc_defs.h [hunk 6]
```diff
@@ -55,10 +55,14 @@
...
-#ifdef _DEBUG
+#if defined(_DEBUG)
     #define ASC_PRINTF(...)     printf(__VA_ARGS__)
     #define ASC_FPRINTF(...)    fprintf(__VA_ARGS__)
     #define ASC_FFLUSH(x)       fflush(x)
+#elif defined(ANDROID)
+	#define ASC_PRINTF(...)
+    #define ASC_FPRINTF(...)
+    #define ASC_FFLUSH(x)
 #else
```
Comment (dvrogozh) [inline _studio/shared/asc/include/asc_defs.h]: Even in DEBUG mode we should not print anything to console in the library. I.e. This code does not make ANY sense actually. This whole block should be rewritten at least as:
```
//#define ASC_DEBUG

#if defined(ASC_DEBUG)
    #define ASC_PRINTF(...)     printf(__VA_ARGS__)
    #define ASC_FPRINTF(...)     fprintf(__VA_ARGS__)
    #define ASC_FFLUSH(x)    fflush(x)
#else
    #define ASC_PRINTF(...)
    #define ASC_FPRINTF(...)
    #define ASC_FFLUSH(x)
#endif
```
Pay attention that ASC_DEBUG is commented out!

Comment (Losev-Ivan) [reply _studio/shared/asc/include/asc_defs.h]: Should this change be made in closed source?

File: _studio/mfx_lib/cmrt_cross_platform/include/cmrt_cross_platform.h [hunk 7]
```diff
@@ -70,6 +73,16 @@ class SurfaceIndex
...
 #ifdef CM_LINUX
+    /*
+     * Do not delete this line:
+     * SurfaceIndex is commonly used as CM kernel function's parameter.
+     * It has virutal table and has copy constructor, so GNU calling convetion will pass the object's pointer to kernel function.
+     * This is different with Windows VC++, which always copies the entire object transferred on the callee's stack.
+     *
+     * Depending on the special object size after adding below "extra_byte",
+     * SetKernelArg and SetThreadArg can recognize this object and follow GNU's convention to construct kernel function's stack.
+     * Detail can be found on "http://wiki.ith.intel.com/display/VPGCompiler/CM+Linux"
```
Comment (dvrogozh) [inline _studio/mfx_lib/cmrt_cross_platform/include/cmrt_cross_platform.h]: Please, remove the link from the comment.

Comment (Losev-Ivan) [reply _studio/mfx_lib/cmrt_cross_platform/include/cmrt_cross_platform.h]: @dvrogozh removed
